### PR TITLE
Implements semigroup and monoid for Maybe.

### PR DESCRIPTION
**Pull request by diasbruno**
Implements semigroup and monoid for Maybe.

### CODE REVIEW

File: src/data/maybe/maybe.js [hunk 1]
```diff
@@ -42,6 +42,10 @@ const assertMaybe = assertType(Maybe);
 
 
 extend(Just.prototype, {
+  empty() {
```
Comment (robotlolita) [inline src/data/maybe/maybe.js]: You can remove this definition.

File: src/data/maybe/maybe.js [hunk 2]
```diff
@@ -52,6 +56,12 @@ extend(Just.prototype, {
   }
 });
 
+extend(Nothing.prototype, {
+  empty() {
```
Comment (robotlolita) [inline src/data/maybe/maybe.js]: And this one.

File: src/data/maybe/maybe.js [hunk 3]
```diff
@@ -174,6 +184,42 @@ adtMethods(Maybe, {
...
+      });
+    }
+  },
+
+
+  /*~
+   * type: |
+   *   forall a: () => Maybe a
+   */
+  empty: {
```
Comment (robotlolita) [inline src/data/maybe/maybe.js]: And this :)

File: src/data/maybe/maybe.js [hunk 4]
```diff
@@ -233,6 +279,15 @@ Object.assign(Maybe, {
 
 
   /*~
+   * type: |
+   *   forall a: () => Maybe a
+   */
+  empty() {
```
Comment (robotlolita) [inline src/data/maybe/maybe.js]: They're all automatically derived from this one definition here :)

**Comment by robotlolita** [comment]
Looks neat, thanks! :)

I've requested a few simple changes and then it's ready to be merged :)

**Comment by diasbruno** [comment]
hmm, I've forgot to remove those when you fixed the test issue. :) 
Is anything else missing? (docs...)

**Comment by robotlolita** [comment]
Nice :+1: 

Could you add an `authors` metadata to the methods you've added? Like:

```
/*~
 * authors:
 *   - "@diasbruno"
 * type: |
 *   ...
 */
```

If you're up to writing docs too, a short description with usage examples ([as explained here](https://github.com/origamitower/folktale/blob/master/docs/source/en/guides/contributing/2-contributing-documentation.md#documenting-a-new-entity)) would be neat! In this case, the category for `empty` would be `Constructing`, and the category for `concat` would be `Combining`.

**Comment by diasbruno** [comment]
I tried to write all the docs, but `metamagical` was not happy and it could not parse the full comment. So this should be ready to merge. 

Hope to take some time to hack with `metamagical` (seems pretty cool).
I'll provide the error I got later.

**Comment by robotlolita** [comment]
No worries :) Thanks

**Pull request merged**